federation/pkg/kubefed/init/init.go

```diff
@@ -163,6 +165,7 @@ func (o *initFederationOptions) Bind(flags *pflag.FlagSet, defaultImage string)
 	flags.StringVar(&o.controllerManagerOverridesString, "controllermanager-arg-overrides", "", "comma separated list of federation-controller-manager arguments to override: Example \"--arg1=value1,--arg2=value2...\"")
 	flags.StringVar(&o.apiServerServiceTypeString, apiserverServiceTypeFlag, string(v1.ServiceTypeLoadBalancer), "The type of service to create for federation API server. Options: 'LoadBalancer' (default), 'NodePort'.")
 	flags.StringVar(&o.apiServerAdvertiseAddress, apiserverAdvertiseAddressFlag, "", "Preferred address to advertise api server nodeport service. Valid only if '"+apiserverServiceTypeFlag+"=NodePort'.")
+	flags.Int32Var(&o.apiServerNodePortsPort, apiserverPortFlag , 0, "Preferred port to use for api server nodeport service (0 for random port assignment). Valid only if '"+apiserverServiceTypeFlag+"=NodePort'.")
```

Comment:
This is correct we cannot do much about it here, since we do not know the port range the admin has set. We need to report back the "out of range" error and we do so here: https://github.com/kubernetes/kubernetes/pull/46283/files#diff-e7243d86cc50e209d78ce51d57dad48dR487